cmd/bnscli/clitests/batch_proposal.test.gold

```diff
@@ -0,0 +1,80 @@
+{
+	"Sum": {
+		"CreateProposalMsg": {
+			"metadata": {
+				"schema": 1
+			},
+			"title": "my proposal",
+			"raw_option": "4gP1AQpLmgNICgIIARIUFNHnXmonj+Fa9Rrz/Xo827xAc1waFIGqiIN1N/rdYKVPZHQC08vYerWbIgcIAhoDSU9WKg1zZW5kaW5nIDIgSU9WCk2aA0oKAggBEhRr63ua555cMBucOwTojh62IuvHBxoU5AUySM5lZoaGRLBD4TKSp/Zb/eUiCAgJGgRET0dFKg5zZW5kaW5nIDkgRE9HRQpLmgNICgIIARIUkfTGalZvv6XGNsXJ2Qn9P0tYeWYaFIEY7PnylIXtLJy1pi4mJehtVVg7IgcIBxoDQlRDKg1zZW5kaW5nIDcgQlRDCgqqAwcKAggBEgEE",
+			"description": "yet another proposal",
+			"election_rule_id": "AAAAAAAAAAM=",
+			"start_time": 1609499460
+		}
+	}
+}
+
+The above transaction is a proposal for executing the following messages:
+{
+	"BatchMsg": {
+		"messages": [
+			{
+				"Sum": {
+					"SendMsg": {
+						"metadata": {
+							"schema": 1
+						},
+						"src": "14D1E75E6A278FE15AF51AF3FD7A3CDBBC40735C",
+						"dest": "81AA88837537FADD60A54F647402D3CBD87AB59B",
+						"amount": {
+							"whole": 2,
+							"ticker": "IOV"
+						},
+						"memo": "sending 2 IOV"
+					}
+				}
+			},
+			{
+				"Sum": {
+					"SendMsg": {
+						"metadata": {
+							"schema": 1
+						},
+						"src": "6BEB7B9AE79E5C301B9C3B04E88E1EB622EBC707",
+						"dest": "E4053248CE6566868644B043E13292A7F65BFDE5",
+						"amount": {
+							"whole": 9,
+							"ticker": "DOGE"
+						},
+						"memo": "sending 9 DOGE"
+					}
+				}
+			},
+			{
+				"Sum": {
+					"SendMsg": {
+						"metadata": {
+							"schema": 1
+						},
+						"src": "91F4C66A566FBFA5C636C5C9D909FD3F4B587966",
+						"dest": "8118ECF9F29485ED2C9CB5A62E2625E86D55583B",
+						"amount": {
+							"whole": 7,
+							"ticker": "BTC"
+						},
+						"memo": "sending 7 BTC"
+					}
+				}
+			},
+			{
+				"Sum": {
+					"ReleaseEscrowMsg": {
+						"metadata": {
+							"schema": 1
+						},
+						"escrow_id": "BA=="
```

Comment:
This is an invalid ID. They are all 8 bytes bigendian integers.
I would suggest that the input takes an integer and encodes it properly.
Or else takes this hex, but enforces it to be 16 chars (if cut and paste from some output)

Reply:
I was wondering about doing the validation of the input. I have decided that accepting most values and letting the server validate it is easier. For the user it is a similar result - the validation fails. At the same time, we do not have to hardcode a custom validation logic into the `bnscli`. `bnscli` is also more powerful as we can create invalid messages to check if `bnsd` validation is correct ;)

What do you think?